app/services/balance_transactions/updater.rb

```diff
@@ -0,0 +1,29 @@
+class BalanceTransactions::Updater
+  Params = Struct.new(:date, :comment, :amount, :transaction_id)
+
+  def initialize(params)
+    @params = Params.new(params[:date], params[:comment],
+      params[:amount], params[:transaction_id])
+  end
+
+  def update
+    ActiveRecord::Base.transaction do
+      update_balance_transaction
+      update_transaction
+    end
+  end
+
+  private
+
+  def transaction
+    @_transaction = Transaction.find(@params[:transaction_id])
+  end
+
+  def update_balance_transaction
+    transaction.transactinable.update(comment: @params[:comment])
+  end
+
+  def update_transaction
+    @_transaction.update(date: @params[:date].to_date, amount: @params[:amount])
```

Comment:
@marderer just `transaction` instead of `@_transaction`?

Reply:
@marderer as we discussed - we read code from up to down, so logic should be organized from up to down. So, `#transition` should go after it's direct call -> after `#update_balance_transaction` and `#update_transaction` 

Let's discuss it in #general if it's still not clear.

Reply:
@AleksSenkou It's clear, sorry, I didn't see it.